### PR TITLE
Update integration test to lower runtimes

### PR DESCRIPTION
**Pull request by simonsan**
Does this have any downsides? It should speed up the integration tests overall?

could also help with #832

related: https://github.com/iqlusioninc/abscissa/issues/845

### CODE REVIEW

File: cargo-audit/tests/acceptance.rs [hunk 1]
```diff
@@ -26,9 +26,9 @@ static ADVISORY_DB_DIR: Lazy<TempDir> = Lazy::new(|| TempDir::new().unwrap());
 /// be multithreaded invocations as `cargo test` executes tests in
 /// parallel by default.
 pub static RUNNER: Lazy<CmdRunner> = Lazy::new(|| {
-    let mut runner = CmdRunner::default();
+    let mut runner = CmdRunner::new(env!("CARGO_BIN_EXE_CARGO_AUDIT"));
```
Comment (tarcieri) [inline cargo-audit/tests/acceptance.rs:29]: What is this environment variable?

Comment (Shnatsel) [reply cargo-audit/tests/acceptance.rs:29]: It seems this should be using `std::env::var_os` instead of `env!`. 

I doubt this is going to be any faster, but I'd be happy to be proven wrong.

Comment (Shnatsel) [reply cargo-audit/tests/acceptance.rs:29]: Panic - it's supposed to be always set: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Comment (Shnatsel) [reply cargo-audit/tests/acceptance.rs:29]: Nevermind, I got it all wrong. It should be `env!(CARGO_BIN_EXE_cargo-audit)`

**Comment by Shnatsel** [comment]
**TL;DR:** this is a worthwhile change, but I'd like to see it on the framework level (i.e. in `abscissa`) rather than fix up every user individually.

I've compared this against the main branch, measuring the second run of the tests (once the crates.io index cloning is done, which takes up the vast majority of the time), and it takes 50 seconds regardless of these changes.

With an up-to-date crates.io index, running `cargo clean; cargo test --release --all-features --no-run` and then measuring `cargo test --release --all-features` shows the new approach to be faster, because `cargo run` is invoked without `--release` and genuinely needs compile the debug build instead of using the release build that's already been compiled.

Contrary to @simonsan's suspicion, the compilation is performed exactly once, and cached after that. There's no recurring compilation assuming the Cargo cache works properly.

**Comment by tarcieri** [comment]
>  I had several cmd mutex poisened errors when running tests with the default configuration of CmdRunner for rustic in the beginning.

This will occur if a test fails while the mutex is held.

It should probably use `panic::catch_unwind` to prevent that.
